server/autolinkplugin/plugin.go

```diff
@@ -142,62 +141,69 @@ func (p *Plugin) ProcessPost(c *plugin.Context, post *model.Post) (*model.Post,
 		}
 	}
 
-	markdown.Inspect(post.Message, func(node interface{}) bool {
-		switch node.(type) {
-		// never descend into the text content of a link/image
-		case *markdown.InlineLink:
-			return false
-		case *markdown.InlineImage:
-			return false
-		case *markdown.ReferenceLink:
-			return false
-		case *markdown.ReferenceImage:
-			return false
+	for _, link := range conf.Links {
```

Comment:
2/5 I am a little concerned about this change, it may need some benchmarking. The new version invoked markdown.Inspect in a loop, and it may be expensive. Is it directly related to the functional change? Can the functional change be implemented in a single `Inspect` pass?

Reply:
Yes, I think this is fundamentally required for this plugin, because each replacement has the capacity to change the markdown structure - therefore if we proceed to look at later replacements, we can no longer apply the rules regarding which markdown elements they can touch.

If you restore the original pattern of inspecting once, the unit tests I have added will break because one replacement inserts a new inline link and a later replacement (which should ignore that inline link) gets erroneously applied because it is still treating it as text.

Reply:
Right. Understood. 1/5 We should benchmark it on a few examples. I wonder if we can re-use the parsed markdown tree between the no-change links.

Reply:
I agree that we should try to re-use parsed markdown or detect the structure change. Otherwise performance will suffer.